src/cobra/flux_analysis/variability.py

```diff
@@ -230,25 +253,43 @@ def flux_variability_analysis(
             model.add_cons_vars([flux_sum, flux_sum_constraint])
 
         model.objective = Zero  # This will trigger the reset as well
-        for what in ("minimum", "maximum"):
-            if processes > 1:
-                # We create and destroy a new pool here in order to set the
-                # objective direction for all reactions. This creates a
-                # slight overhead but seems the most clean.
-                chunk_size = len(reaction_ids) // processes
-                with ProcessPool(
-                    processes,
-                    initializer=_init_worker,
-                    initargs=(model, loopless, what[:3]),
-                ) as pool:
-                    for rxn_id, value in pool.imap_unordered(
-                        _fva_step, reaction_ids, chunksize=chunk_size
-                    ):
+
+        _Progress = Progress
+        if logger.level != logging.INFO:
+            _Progress = _FakeProgress
+
+        with _Progress() as progress:
+            progress_task = progress.add_task(
+                "[cyan]Performing FVA...", total=num_reactions
+            )
+
+            for what in ("minimum", "maximum"):
+                if processes > 1:
+                    # We create and destroy a new pool here in order to set the
+                    # objective direction for all reactions. This creates a
+                    # slight overhead but seems the most clean.
+                    chunk_size = len(reaction_ids) // processes
+                    with ProcessPool(
+                        processes,
+                        initializer=_init_worker,
+                        initargs=(model, loopless, what[:3]),
+                    ) as pool:
+                        for i, (rxn_id, value) in enumerate(
+                            pool.imap_unordered(
+                                _fva_step, reaction_ids, chunksize=chunk_size
+                            )
+                        ):
+                            fva_result.at[rxn_id, what] = value
+                            _update_progress_bar(
+                                progress, progress_task, (i + 1) / num_reactions
```

Comment:
I haven't tested this but does this actually show the right numbers? Like the bar is set to a maximum of nr steps but then you do 2*nr updates. Once for the min and once for the max...

Reply:
Yep, agreed. Didn't mention it now because it seems like we want a different design anyway. Also, use `enumerate(..., start=1)` instead of adding + 1.